onmt/inputters/inputter.py

```diff
@@ -304,7 +304,20 @@ def build_vocab(train_dataset_files, fields, data_type, share_vocab,
 
     # Load vocabulary
     src_vocab = load_vocabulary(src_vocab_path, tag="source")
+    if src_vocab is not None:
+        src_vocab_size = len(src_vocab)
+        logger.info('Loaded source vocab has %d tokens.' % src_vocab_size)
+        for i, token in enumerate(src_vocab):
+            # keep the order of tokens specified in the vocab file by
+            # adding them to the counter with decreasing counting values
+            counter['src'][token] = src_vocab_size - i
```

Comment:
It seems a bit arbitrary to me. Could you elaborate why you think it's necessary/relevant?

Reply:
If we only count the words in the corpus that are present in the vocabulary file we will end up with a vocabulary in the Counter object that not only may be smaller than the vocab in the specified file but also that is in a completely different order. Keeping the order and size of a specified vocabulary file can be useful, for instance, in semi-supervised settings (e.g. fusion techniques - https://arxiv.org/pdf/1503.03535.pdf) where the output softmax of a model needs to be aligned with the output softmax of a pretrained language model.

What I did there in the code is a hack, of course. I'm artificially giving descending counts to the tokens in the vocabulary list so that when torchtext's Vocab is created the order of the specified vocabulary file is maintained.